### PR TITLE
Add style checkers

### PR DESCRIPTION
**Pull request by original-brownbear**
WIP, it's very hard to get in compliance with Elasticsearch's rules in our codebase.

### CODE REVIEW

File: buildSrc/src/main/resources/checkstyle.xml [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        -->
+
+        <!-- Non-inner classes must be in files that match their names. -->
+        <module name="OuterTypeFilename" />
+        <!-- No line wraps inside of import and package statements. -->
+        <module name="NoLineWrap" />
+        <!-- Each java file has only one outer class -->
+        <module name="OneTopLevelClass" />
+        <!-- The suffix L is preferred, because the letter l (ell) is often
+        hard to distinguish from the digit 1 (one). -->
```
Comment (jordansissel) [inline buildSrc/src/main/resources/checkstyle.xml]: The comment here plus the name of this check "UpperEll" was not enough to inform me about the behavior. Googling around in search of understanding, this seems to be about requiring an uppercase "L" specifically on long literals? Like "1000L" vs "1000l"

Can you add in the comment that this is for `long` literals?

File: buildSrc/src/main/resources/checkstyle.xml [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            <property name="format" value="serialVersionUID" />
+            <property name="message" value="Do not declare serialVersionUID." />
+            <property name="ignoreComments" value="true" />
+        </module>
+        <module name="RegexpSinglelineJava">
+            <property name="format" value="java\.io\.Serializable" />
+            <property name="message" value="References java.io.Serializable." />
+            <property name="ignoreComments" value="true" />
+        </module>
+        <!-- end Orwellian suppression of Serializable -->
```
Comment (jordansissel) [inline buildSrc/src/main/resources/checkstyle.xml]: > Orwellian suppression of

This description is confusing.  Maybe omit these three words?

File: buildSrc/src/main/resources/checkstyle.xml [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+    <!-- Checks Java files and forbids empty Javadoc comments -->
+    <module name="RegexpMultiline">
+        <property name="format" value="\/\*[\s\*]*\*\/"/>
+        <property name="fileExtensions" value="java"/>
+        <property name="message" value="Empty javadoc comments are forbidden"/>
+    </module>
+
+    <module name="TreeWalker">
+        <!-- Its our official line length! See checkstyle_suppressions.xml for the files that don't pass this. For now we
```
Comment (jordansissel) [inline buildSrc/src/main/resources/checkstyle.xml]: > checkstyle_suppressions.xml 

Should there be checkstyle_suppressions.xml added in this PR? I do not see one.

**Comment by jordansissel** [comment]
If an IntellIJ config for this style accompanies this PR, it would be cool. @original-brownbear thoughts on this? It's OK if this is out of scope.

**Comment by original-brownbear** [comment]
@jordansissel wops I totally forgot to finish this one :) Actually yea we need suppressions and I was planing on reverting the code changes for now so we can go one by one. Anything else is just too crazy :) The rules are taken 1:1 from ES and imo it's worth it to not adjust them, simply to stay in line with what they do right?

**Comment by original-brownbear** [comment]
@jordansissel I'm not 100% sure on this, but if I remember correctly Idea is able to understand code style configs without any additional configuration (not sure how well that works, I know for scalastyle it works very well though :)).

**Comment by jordansissel** [comment]
>  I was planing on reverting the code changes for now

++ I am personally using the code changes mostly as a "This is what the code looks like after style checks pass". I am OK if this PR only includes the checkstyle.xml and omits the code changes which we could do at a later time.

> Idea is able to understand code style configs without any additional configuration 

Oh! Delightful. We can make this work -- no change required for this PR on Intellij/IDEA

> The rules are taken 1:1 from ES and imo it's worth it to not adjust them

My comments are generally about the way we describe the rules (the comments in the xml file), so I'm OK if we change them, and we can follow up with a PR on elasticsearch to make the same clarity changes there. Thoughts?

**Comment by jordansissel** [comment]
That said, I'm OK if we copy Elasticsearch's for now and improve the internal commenting later.

**Comment by original-brownbear** [comment]
@jordansissel alrighty reverted all code changes and excluded everything (will have to make the exclude less coarse in the next step(s)) now.
I think this is a valid step 0 here :)

**Comment by original-brownbear** [comment]
Jenkins test this please

File: buildSrc/src/main/resources/checkstyle.xml [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+          suppress the check there but enforce it everywhere else. This prevents the list from getting longer even if it is
+          unfair. -->
+        <module name="LineLength">
+            <property name="max" value="100"/>
```
Comment (danhermann) [inline buildSrc/src/main/resources/checkstyle.xml]: Given the choice, I'd opt for longer lines. I think it's a pretty rare screen these days that won't accommodate line lengths of 120 or more.

Comment (original-brownbear) [reply buildSrc/src/main/resources/checkstyle.xml]: @danhermann yea but diffing (and eventually reading too) becomes a pain with more and more information per line :) => I'm a big fan of `100` here tbh :D

Comment (danhermann) [reply buildSrc/src/main/resources/checkstyle.xml]: Sure, that's fine. I just threw it out there in case no one had other preferences.

**Comment by original-brownbear** [comment]
Jenkins test this please

**Comment by jakelandis** [comment]
can we wait to merge this till after right after 6.2 (or 6.3) branch is cut?  it has the potential to add alot of noise to a pending feature freeze. 

Also ++ to longer lines.

**Comment by kares** [comment]
this seem to have never materialized, I think this effort (if needed) should be restarted with a clean slate

**Pull request closed without merging**